Packs/Jira/Integrations/JiraV3/JiraV3.py

```diff
@@ -0,0 +1,4381 @@
+        super().__init__(base_url=base_url, proxy=proxy, verify=verify, headers=headers)
+
+    @abstractmethod
+    def test_instance_connection(self) -> None:
+        """This method is used to test the connectivity of each instance, each child will implement
+        their own connectivity test
+        """
+        pass
+
+    def http_request_with_access_token(self, method, headers: Dict[str, str] | None = None, url_suffix='', params=None, data=None,
+                                       json_data=None, resp_type='json', ok_codes=None, full_url='',
+                                       files: Dict[str, Any] | None = None) -> Any:
+        """This method wraps the _http_request that comes from the BaseClient class, and adds the access_token
+        of the client to the headers request, by calling the get_access_token method, which is an abstract method,
+        and every child class mush implement it.
+        Returns:
+            _type_: _description_
+        """
+        if headers is None:
+            headers = {}
+        access_token = self.get_access_token()
+        # We unite multiple headers (using the '|' character, pipe operator) since some requests may require extra headers
+        # to work, and this way, we have the option to receive the extra headers and send them in the API request.
+        request_headers = self._headers | headers | {'Authorization': f'Bearer {access_token}'}
+        return self._http_request(method, url_suffix=url_suffix, full_url=full_url, params=params, data=data,
+                                  json_data=json_data, resp_type=resp_type, ok_codes=ok_codes, files=files,
+                                  headers=request_headers)
+
+    # Authorization methods
+    def get_access_token(self) -> str:
+        # CONFLUENCE Explain the process of saving and retrieving the access token from the integration's context
+        """This function is in charge of returning the access token stored in the integration's context. If the access token
+        has expired, we try to retrieve another access token using a refresh token that is configured in the integration's context
+
+        Raises:
+            DemistoException: If no access token was configured.
+            DemistoException: If no refresh token was configured.
+
+        Returns:
+            str: The access token to send with the requests.
+        """
+        integration_context = get_integration_context()
+        token = integration_context.get('token', '')
+        if not token:
+            raise DemistoException(('No access token was configured, please complete the authorization process'
+                                    ' as shown in the documentation'))
+        # The valid_until key stores the valid date in seconds to make it easier for comparison
+        valid_until = integration_context.get('valid_until', 0)
+        current_time = get_current_time_in_seconds()
+        if current_time >= valid_until:
```

Comment:
I think we should add this - let's discuss that.
```suggestion
        if current_time >= valid_until - 10:
```